File: testdata/minimal-testbeds/scala/src/Greeter.scala
package demo

import scala.util.Try
import demo.helpers.Format

object Greeter {
  def greet(name: String): String = {
    format(name)
  }

  def format(s: String): String = s.toUpperCase
}

trait Auditable

class BaseGreeter

class LoggedGreeter extends BaseGreeter with Auditable {
  def loud(name: String): String = {
    Greeter.greet(name)
  }
}

File: testdata/minimal-testbeds/elixir/lib/greeter.ex
defmodule Demo.Greeter do
  alias Demo.Format
  import Demo.Helpers
  use GenServer
  @behaviour GenServer

  @doc "Public greet entry — pipe normalize then Format.apply."
  def greet(name) do
    name
    |> normalize()
    |> Format.apply()
  end

  def start_link(opts \\ []) do
    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
  end

  @impl true
  def init(opts), do: {:ok, opts}

  defp normalize(name) when is_binary(name), do: String.trim(name)
end
